In [1]:
import numpy as np

In [2]:
fh = open("newtons-rings-data.txt", "r")
data = fh.read().splitlines()
data = np.array([np.array(list(map(float, row.split()))) for row in data])

In [3]:
m = 10
wavl = 589e-9

In [4]:
D = (data[:,3] - data[:,6])**2 / 1e6
R = np.array([(D[i] - D[len(D)-m])/(4*(data[i,0] - m)*wavl) for i in range(len(D) - m)])
R

array([1.54665535, 1.56805791, 1.55574597, 1.56589255, 1.55398981,
       1.56040747, 1.58225806, 1.5381296 , 1.51330645, 1.55857385])

In [5]:
print(f"{np.average(R):.2f}\t{np.std(R, ddof=1):.2f}")

1.55	0.02


Hence the radius of curvature of the lens was obtained to be $(1.55 \pm 0.02)$ m